mypy/stubdoc.py

```diff
@@ -20,7 +20,7 @@
 Sig: _TypeAlias = Tuple[str, str]
 
 
-_TYPE_RE: Final = re.compile(r"^[a-zA-Z_][\w\[\], .\"\']*(\.[a-zA-Z_][\w\[\], ]*)*$")
+_TYPE_RE: Final = re.compile(r"^[a-zA-Z_][\w\[\], ().\"\']*(\.[a-zA-Z_][\w\[\], ]*)*$")
```

Comment:
So basically this giant regex matches these:

- a
- abc
- a[1], b(2).c
- a_b.c_d
- a.b.c